api/graphite.go

```diff
@@ -132,8 +132,11 @@ func (s *Server) findSeriesLocal(ctx context.Context, orgId int, patterns []stri
 		defer span.Finish()
 		nodes, err := s.MetricIndex.Find(orgId, pattern, seenAfter)
 		if err != nil {
-			tags.Error.Set(span, true)
-			return nil, response.NewError(http.StatusBadRequest, err.Error())
+			err := response.WrapError(err)
+			if err.Code() != http.StatusBadRequest {
+				tags.Error.Set(span, true)
```

Comment:
this seems to be the same like the below used `tracing.Failure(span)` or not? for consistency i'd use either one or the other only, otherwise it's confusing because the reader first needs to figure out that they're the same

Reply:
you're right